### PR TITLE
[ci] Limit num gpus if specified for A100

### PR DESCRIPTION
**Pull request by khluu**

### CODE REVIEW

File: .buildkite/test-template-aws.j2 [hunk 1]
```diff
@@ -75,7 +75,7 @@ steps:
             - "'cd {{ (step.working_dir or default_working_dir) | safe }} && {{ step.command or (step.commands | join(' && ')) | safe }}'"
             resources:
               limits:
-                nvidia.com/gpu: 8
+                nvidia.com/gpu: {{ step.num_gpus or 8 }}
```
Comment (youkaichao) [inline .buildkite/test-template-aws.j2:78]: should it be 1 by default?

**Review by youkaichao — COMMENTED** [review]
would you mind update the number of gpus for custom allreduce test introduced in https://github.com/vllm-project/vllm/pull/5689 as well? it needs 4 gpus.

File: .buildkite/test-template-aws.j2 (hunk 1 above)
Comment (khluu) [reply .buildkite/test-template-aws.j2]: ah yes. updated to 1

**Comment by khluu** [comment]
> would you mind update the number of gpus for custom allreduce test introduced in https://github.com/vllm-project/vllm/pull/5689 as well? it needs 4 gpus.

done

**Review by youkaichao — APPROVED** [review]
thanks for the quick response! please also check if the nightly benchmark needs to adjust `num_gpus` .

File: .buildkite/test-template-aws.j2 [hunk 2]
```diff
@@ -75,7 +75,7 @@ steps:
             - "'cd {{ (step.working_dir or default_working_dir) | safe }} && {{ step.command or (step.commands | join(' && ')) | safe }}'"
             resources:
               limits:
```
Comment (youkaichao) [inline .buildkite/test-template-aws.j2:77]: one more nit comment: is it necessary that we duplicate part of code here? is it possible to reuse the existing code, and only modify say `queue` based on the `step.gpu` value?

**Comment by youkaichao** [comment]
I merged it so that we can run a100 tests in parallel. please check the rest comments later @khluu

File: .buildkite/test-template-aws.j2 (hunk 2 above)
Comment (khluu) [reply .buildkite/test-template-aws.j2:77]: Our A100 is set up in the EKS cluster so it needs the k8s plugin here to configure the pod (see `podSpec`) in the cluster that runs the job. Other jobs run in a Docker container inside an individual EC2 instance controlled by AWS autoscaling group so it's using Docker plugin.

Comment (khluu) [reply .buildkite/test-template-aws.j2:77]: Since the plugins used are different, I don't think we can reuse most of the code here

**Pull request merged**